File: taskmaster-pro-ui/src/app/shared/models/schedule.ts
import { UserDto } from './user.dto';

export interface ScheduleDto {
  id: string;
  orderId: string;
  title: string;
  scheduledStart: string;
  scheduledEnd: string;
  description: string;
  assignedTo?: UserDto;
  created: string;
  createdBy: string;
  updated?: string;
  updatedBy?: string;
}

export interface CreateScheduleDto {
  orderId: string;
  title: string;
  scheduledStart: string;
  scheduledEnd: string;
  description: string;
  assignedToId: string;
}

export interface UpdateScheduleDto {
  id: string;
  orderId: string;
  title: string;
  scheduledStart: string;
  scheduledEnd: string;
  description: string;
  assignedToId: string;
}

export interface ScheduleDetailDto {
  id: string;
  orderId: string;
  title: string;
  scheduledStart: string;
  scheduledEnd: string;
  description: string;
  assignedTo?: UserDto;
  created: string;
  createdBy?: UserDto;
  updated?: string;
  updatedBy?: UserDto;
}

File: taskmaster-pro.Domain/Entities/Schedule.cs
namespace taskmaster_pro.Domain.Entities
{
    public class Schedule
    {
        public Guid Id { get; set; }
        public string Title { get; set; }
        public Guid OrderId { get; set; }
        public DateTime ScheduledStart { get; set; }
        public DateTime ScheduledEnd { get; set; }
        public string Description { get; set; }

        public string? AssignedToId { get; set; }
        public string UserId { get; set; }

        public DateTime Created { get; set; }
        public string CreatedBy { get; set; }
        public DateTime? Updated { get; set; }
        public string? UpdatedBy { get; set; }

        public Order Order { get; set; }
    }
}